packages/x-license-pro/src/useLicenseVerifier/useLicenseVerifier.ts

```diff
@@ -8,16 +8,26 @@ import {
 } from '../utils/licenseErrorMessageUtils';
 import { LicenseStatus } from '../utils/licenseStatus';
 
+let sharedLicenseStatus: LicenseStatus | undefined;
+
 export function useLicenseVerifier(): LicenseStatus {
   return React.useMemo(() => {
-    const newLicenseStatus = verifyLicense(LicenseInfo.getReleaseInfo(), LicenseInfo.getKey());
-    if (newLicenseStatus === LicenseStatus.Invalid) {
+    if (sharedLicenseStatus !== undefined) {
```

Comment:
Beyond checking if there's a status saved you could also check if the current `LicenseInfo.getKey()` matches the one used when `sharedLicenseStatus` was set. This prevents the user from changing the key from a valid to an invalid and don't see the watermark.

Reply:
Done